packages/apps/plugins/plugin-table/src/components/TableMain.tsx

```diff
@@ -2,30 +2,30 @@
 // Copyright 2023 DXOS.org
 //
 
-import React, { type FC, useRef } from 'react';
+import React, { type FC } from 'react';
 
 import { Main } from '@dxos/react-ui';
+import { Table } from '@dxos/react-ui-table';
 import { baseSurface } from '@dxos/react-ui-theme';
 
 import { ObjectTable, type ObjectTableProps } from './ObjectTable';
 
-const TableMain: FC<ObjectTableProps> = ({ table }) => {
-  const containerRef = useRef<HTMLDivElement | null>(null);
-
-  return (
-    <Main.Content
-      classNames={[baseSurface, 'fixed inset-inline-0 block-start-[--topbar-size] block-end-0 overflow-auto']}
-      ref={containerRef}
-    >
-      <ObjectTable
-        table={table}
-        key={table.id} // New component instance per table.
-        stickyHeader
-        role='grid'
-        getScrollElement={() => containerRef.current}
-      />
-    </Main.Content>
-  );
-};
+const TableMain: FC<Omit<ObjectTableProps, 'getScrollElement'>> = ({ table }) => (
+  <Main.Content>
+    <Table.Root>
+      {/* TODO(Zan): Is moving the classes from Main.Content to the viewport ok? */}
+      <Table.Viewport
+        classNames={[baseSurface, 'fixed inset-inline-0 block-start-[--topbar-size] block-end-0 overflow-auto']}
+      >
```

Comment:
This won’t work in Deck, but that’s not merged yet.

An approach that’d be consistent with the rest of the UI business would be:
```tsx
<Table.Root>
  <Main.Content asChild>
    <Table.Viewport {...etc} />
```

Reply:
This _should_ merge `classNames` for `Table.Viewport` and `Main.Content`, so you wouldn’t need to repeat them here.